File: tests/contexts/graph/notes/domain/note_mother.py
import json

from src.contexts.graph.notes.domain.note import Note


class NoteMother:
    @classmethod
    def create(cls, number: int) -> Note:
        with open("notes.json") as file:
            note = json.load(file)["results"][number]
        return Note.create(
            id_=note["id"],
            url=note["url"],
            title=note["properties"]["Name"]["title"][0]["plain_text"],
            related_notes=note["properties"]["Related Notes"],
        )

File: backend/contexts/notes/src/test/kotlin/com/dimanupy/notes/note/domain/NoteMother.kt
package com.dimanupy.notes.note.domain

object NoteMother {

    fun anyPrimitives(): NotePrimitives = NotePrimitives(
        notionId = NotionNoteIdPrimitivesMother.create(),
        title = NoteTitlePrimitivesMother.create(),
        url = NoteUrlPrimitivesMother.create(),
        relatedNotes = NoteRelatedNotesPrimitivesMother.create(),
    )

    fun create(
        notionId: String = NotionNoteIdPrimitivesMother.create(),
        url: String = NoteUrlPrimitivesMother.create(),
        title: String = NoteTitlePrimitivesMother.create(),
        relatedNotes: List<String> = NoteRelatedNotesPrimitivesMother.create(),
    ): Note = Note.fromPrimitives(
        NotePrimitives(
        notionId = notionId,
        title = title,
        url = url,
        relatedNotes = relatedNotes,
        )
    )
}
